gnosis/safe-react
# Display readable Safe contract errors

## What it solves
Resolves #2804

## How this PR fixes it
Errors returned from the contract are verified against the `CONTRACT_ERRORS` map and, if found, displayed in a snackbar notification.

A map of errors codes-messages was added and a decoding function created. If an if found, it returns `[CODE]: [MESSAGE]` otherwise the message. The `notification` object in `createTransaction`/`processTransaction` was moved to facilitate adding the returned error.

## How to test it
Trigger a contract-level error and await a notification which should contain the formatted message, i.e.

1. Enable module
2. Try to enable the same module again

## Screenshots
![image](https://user-images.githubusercontent.com/20442784/139399009-ac490531-4d5a-40de-a13d-5061afbd675a.png)
